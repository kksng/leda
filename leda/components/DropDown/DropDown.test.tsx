import React from 'react';
import {
  render, screen,
} from '@testing-library/react';
import { DropDown } from './index';

describe('DropDown SNAPSHOTS', () => {
  it('should render', () => {
    const wrapper = render((
      <DropDown />
    ));

    expect(wrapper.container).toMatchSnapshot();
  });
});

describe('DropDown ATTRIBUTES', () => {
  describe('className prop', () => {
    it('should have className given by prop', () => {
      render((
        <DropDown _box>test</DropDown>
      ));

      expect(screen.queryByText('test')).toHaveClass('box');
    });

    it('should change classes through props', () => {
      render((
        <DropDown _active>test</DropDown>
      ));

      expect(screen.queryByText('test')).not.toHaveClass('box');

      expect(screen.queryByText('test')).toHaveClass('active');
    });

    it('className should not change prop-classes', () => {
      render((
        <DropDown _active className="test">test</DropDown>
      ));

      expect(screen.queryByText('test')).not.toHaveClass('box');

      expect(screen.queryByText('test')).toHaveClass('active');

      expect(screen.queryByText('test')).toHaveClass('test');
    });
  });

  it('should be wrapped in wrapper', () => {
    const wrapper = render((
      <DropDown />
    ));

    wrapper.rerender((
      <DropDown
        wrapperRender={() => (
          <span />
        )}
      />
    ));

    expect(document.querySelector('span')).toBeInTheDocument();

    wrapper.rerender((
      <DropDown
        wrapperRender={() => (
          <div />
        )}
      />
    ));

    expect(document.querySelector('div')).toBeInTheDocument();

    wrapper.rerender((
      <DropDown
        wrapperRender={() => (
          <p />
        )}
      />
    ));

    expect(document.querySelector('p')).toBeInTheDocument();
  });

  it('should have children prop', () => {
    render((
      <DropDown>
        <div className="level-1"><span className="level-2">test</span></div>
      </DropDown>
    )).debug();

    expect(screen.getByText('test').textContent).toEqual('test');

    expect(screen.getByText('test').tagName).toEqual('SPAN');

    expect(screen.getByText('test').className).toEqual('level-2');

    expect(screen.getByText('test').parentElement?.tagName).toEqual('DIV');

    expect(screen.getByText('test').parentElement?.className).toEqual('level-1');
  });
});
