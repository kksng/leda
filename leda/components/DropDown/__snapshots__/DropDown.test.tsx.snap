// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown SNAPSHOTS should render 1`] = `
<div>
  <span
    class="dropdown-wrapper"
  />
</div>
`;
